pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -199,19 +325,105 @@ func sendCertsProtobuf(attestReq *attest.ZAttestReq, iteration int) {
+		zedcloud.SetDeferred("attest:"+reqID, buf, size, attestURL,
 			zedcloudCtx, true)
 	}
 }
+
+// initialize cipher pubsub trigger handlers and channels
+func cipherModuleInitialize(ctx *zedagentContext, ps *pubsub.PubSub) {
+
+	// create the trigger channels
+	ctx.cipherCtx.triggerEveNodeCerts = make(chan struct{}, 1)
+	ctx.cipherCtx.triggerControllerCerts = make(chan struct{}, 1)
+}
+
+// start the task threads
+func cipherModuleStart(ctx *zedagentContext) {
+	if !zedcloud.UseV2API() {
+		log.Infof("V2 APIs are still not enabled")
+		// we will run the tasks for watchdog
+	}
+	// start the eve node certificate push task
+	go eveNodeCertsTask(ctx, ctx.cipherCtx.triggerEveNodeCerts)
+
+	// start the controller certificate fetch task
+	go controllerCertsTask(ctx, ctx.cipherCtx.triggerControllerCerts)
+}
+
+// Controller certificate, check whether there is a Sha mismatch
+// to trigger the post request
+func handleControllerCertsSha(ctx *zedagentContext,
+	config *zconfig.EdgeDevConfig) {
+	sumHash := hex.EncodeToString(ctx.cipherCtx.cfgControllerCertHash)
+	certHash := config.GetControllercertConfighash()
+	if sumHash != certHash {
+		triggerControllerCertEvent(ctx)
+	}
+}
+
+// Eve node certificate, check whether there is a Sha mismatch
+// to trigger the post request
+func handleEveNodeCertsSha(ctx *zedagentContext,
+	config *zconfig.EdgeDevConfig) {
+	sumHash := hex.EncodeToString(ctx.cipherCtx.eveNodeCertHash)
+	certHash := config.GetNodecertConfighash()
+	if sumHash != certHash {
+		triggerEveNodeCertEvent(ctx)
+	}
+}
+
+//  controller certificate pull trigger function
+func triggerControllerCertEvent(ctxPtr *zedagentContext) {
+
+	log.Info("Trigger for Controller Certs")
+	select {
+	case ctxPtr.cipherCtx.triggerControllerCerts <- struct{}{}:
+		// Do nothing more
+	default:
+		log.Warnf("Failed to send on triggerControllerCerts")
+	}
+}
+
+//  eve node certificate post trigger function
+func triggerEveNodeCertEvent(ctxPtr *zedagentContext) {
+
+	log.Info("Trigger Eve Node Certs publish")
+	select {
+	case ctxPtr.cipherCtx.triggerEveNodeCerts <- struct{}{}:
+		// Do nothing more
+	default:
+		log.Warnf("Failed to send on triggerEveNodeCerts")
+	}
+}
+
+func convertLocalToApiHashAlgo(algo types.CertHashType) evecommon.HashAlgorithm {
+	switch algo {
+	case types.CertHashTypeSha256First16:
+		return evecommon.HashAlgorithm_HASH_ALGORITHM_SHA256_16BYTES
+	default:
+		return evecommon.HashAlgorithm_HASH_ALGORITHM_INVALID
+	}
+}
+
+func convertLocalToApiCertType(certType types.CertType) zcert.ZCertType {
+	switch certType {
+	case types.CertTypeOnboarding:
+		return zcert.ZCertType_CERT_TYPE_DEVICE_ONBOARDING
+	case types.CertTypeRestrictSigning:
+		return zcert.ZCertType_CERT_TYPE_DEVICE_RESTRICTED_SIGNING
+	case types.CertTypeEk:
+		return zcert.ZCertType_CERT_TYPE_DEVICE_ENDORSEMENT_RSA
+	case types.CertTypeEcdhXchange:
+		return zcert.ZCertType_CERT_TYPE_DEVICE_ECDH_EXCHANGE
+	default:
+		return zcert.ZCertType_CERT_TYPE_CONTROLLER_NONE
```

Comment:
Since this is local to API it would be more helpful if this was a log.Fatal so when a new cert type is added it is more obvious that a case is missing. The log.Fatal should include the %d of the value of the certType.